### ISSUE TITLE
Error - "Unexpectedly got an EOF when reading the request"

### ISSUE THREAD

**Issue by rolivieri** [IBM-Swift/Kitura-CouchDB#58]
While using `Kitura-CouchDB` and executing REST requests through its API, we are running into the following issue: 

```
[ClientRequest.swift:360 end(close:)] ClientRequest error. Failed to parse response. Error=Unexpectedly got an EOF when reading the request.
```

I have created a very small program that can be used to easily and consistently reproduce this problem without effort. I am about to create a repo with this sample/test program and share here the steps for reproducing the issue.

**Comment by rolivieri** [comment]
Here are the steps for reproducing the error:

1) Create an instance of Cloudant DB on Bluemix and get its credentials.

2) Clone this repo: `https://github.com/rolivieri/db-test`.

3) Open `DatabaseManager.swift` (`https://github.com/rolivieri/db-test/blob/master/Sources/DatabaseManager.swift`) and update it with the Cloudant DB credentials you obtained in step #1 above.

4) Let’s first build the application in `debug` mode: `swift build --configuration debug`

5) Run the application under the `.build/debug` folder and things should work just fine… you can run it multiple times if you’d like.

6) Now lets build the application in `release` mode: `swift build --configuration release`

7) Run the application under the `.build/release` folder and things **WON’T** work (you can run it multiple times if you’d like):

```
[2017-06-20T07:09:02.896-05:00] [ERROR] [ClientRequest.swift:360 end(close:)] ClientRequest error. Failed to parse response. Error=Unexpectedly got an EOF when reading the request
```


Summary: Running the binaries that were built in `debug` mode works just fine… however, running the binaries that were built in `release` mode ends up in a failure. You can switch back and forth (i.e. running the debug binary and running the release binary) and you should see how one consistently works fine while the other fails.

We will need someone to debug this further beyond what I have done so far and determine if we have an issue in Kitura-CouchDB, or in Swift, or in both… this could be an issue in Swift that may be affecting outbound connections (not just Kitura-CouchDB… but need further investigation to make this determination). If it is an issue in Swift, then a corresponding JIRA issue should be opened.

**Comment by rolivieri** [comment]
Also, just fyi, this issue is reproducible on both platforms, macOS (v10.12.5) and Ubuntu Trusty (v14.04).

**Comment by rolivieri** [comment]
Just fyi to help debug this problem, here is additional info since we are now seeing this problem occur in other applications too. This could be an issue in Swift and/in one of the following libraries used by Kitura-CouchDB:

```
Package          Last version known to work
CCurl            0.2.3
BlueCryptor    0.8.10
Kitura-net       1.7.4
BlueSocket     0.12.45
OpenSSL        0.3.1
```

**Comment by rolivieri** [comment]
@jkingoliver and I have narrowed it down to the following package: CCUrl. When using the latest version of this package (`v0.2.4`), the error above happens consistently if the application was built in `release` mode. However, if we use the previous version of CCurl (`v0.2.3`), the application runs fine even if it was compiled in `release` mode.

This seems to indicate that the change in the following PR (https://github.com/IBM-Swift/CCurl/pull/11) is not getting along with the latest version of Swift... given that the error does **not**  occur if the application was built in `debug` mode. We will need someone with a good understanding of the change made to CCurl to make that analysis.

**Comment by shmuelk** [comment]
@rolivieri  and @jkingoliver thank you very much for the analysis. We will look into this in the morning.

**Comment by na-gupta** [comment]
This has been fixed in CCurl 0.2.5

**Comment by rolivieri** [comment]
Thanks @na-gupta for the quick turnaround. I just tested the fix as well and it worked just fine. Closing this issue.
